main.c

```diff
@@ -221,12 +218,16 @@ asm_return_t lz_main(void)
 	size = (*(u32*)((u8*)zero_page + BP_SYSSIZE)) << 4;
 
 	if (tpm->family == TPM12) {
+		SHA1_CONTEXT sha1ctx;
```

Comment:
I've deliberately changed it to static as it didn't fit in stack otherwise. However, from the bloat-o-meter it seems like increasing stack size would be better. I'll check on hardware how much stack it really needs.

Reply:
Ok, now that the LZ header is moved there are still 0x34 untouched bytes on the stack, no increase required for now.

Reply:
So after having considered this a bit more, I don't think the current sha1 interface is sensible.  Having the full context outside of the translation unit inhibits some pointer based optimisations.  (In particular, the compiler can't determine that the input pointer and the context pointer don't alias.)

It would be better to have an interface similar to sha256, where the internal context is private to the sum function, and only the resulting hash gets written back out to the caller.

I'll see about making this happen in a followup PR